src/arch/xtensa/include/arch/atomic.h

```diff
@@ -44,12 +48,28 @@ static inline int32_t arch_atomic_add(atomic_t *a, int32_t value)
 		: "=&a" (result), "=&a" (current)
 		: "a" (&a->value), "a" (value)
 		: "memory");
-
 	return current;
+#elif XCHAL_HAVE_EXCLUSIVE
+	/*reference xtos : xipc_misc.h*/
+	int32_t result = 0;
+	int32_t current;
+
+	while (!result) {
+		current = XT_L32EX((int32_t *)a);
+		result = current + value;
+		XT_S32EX(result, (int32_t *)a);
+		XT_GETEX(result);
+	}
+	return current;
+#else
+	// Not support atomic
```

Comment:
This is not a problem on unicore UP systems, we  can just do 
```c
#elif CONFIG_NUM_CORES == 1
/* This is fine on UP systems */
*a += value;
#else 
#error No atomic methods available.
#endif
```
same for the others.